### PR TITLE
Add data connectors polling ccp api support

### PR DESCRIPTION
**Pull request by alondanoch**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by anat-gilenson — APPROVED** [review]

**Comment by anat-gilenson** [comment]
> You are updating an existing API version, even though is new, it has to go through the breaking change process, please review the breaking change policy checklist and submit a review.

Thank you for the review @jorgecotillo. 
"2021-09-01-preview" doesn't exist yet and is currently under development. Alon is adding these changes to a development branch. As far as I understand, adding new endpoints to a new API isn't considered breaking change and doesn't need breaking change review.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/dataConnectors.json [hunk 1]
```diff
@@ -1798,6 +1913,325 @@
...
+    },
+    "CodelessConnectorPollingPagingProperties": {
+      "description": "Describe the properties needed to make a pagination call",
+      "type": "object",
+      "properties": {
+        "pagingType": {
+          "description": "Describes the type. could be 'None', 'PageToken', 'PageCount', 'TimeStamp'",
+          "type": "string"
+        },
+        "nextPageParaName": {
```
[the remark this answers is not in the document]
Comment (alondanoch) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/dataConnectors.json:2059]: it's describing the name of the next page parameter... IMO it's a descriptive name

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/dataConnectors.json [hunk 2]
```diff
@@ -1798,6 +1907,325 @@
...
+        "requestConfigUserInputValues": {
+          "type": "array",
+          "description": "",
+          "items": {
+            "type": "object"
+          }
+        }
+      }
+    },
+    "DataConnectorDisconnectBody": {
```
[the remark this answers is not in the document]
Comment (alondanoch) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/dataConnectors.json]: removed

**Comment by alondanoch** [comment]
@jorgecotillo  @filizt - thanks for the review, I have resolved the comments. please review and approve again.

File: specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/dataConnectors.json [hunk 3]
```diff
@@ -288,6 +297,99 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/dataConnectors/{dataConnectorId}/connect": {
+      "post": {
```
Comment (jorgecotillo) [inline specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/dataConnectors.json:301]: >post [](http://example.com/codeflow?start=7&length=4)

question on both of your POST APIs, are they async or sync? if async, you need to follow this:

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#call-action-post-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#call-action-post-asynchronously)

Comment (alondanoch) [reply specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/dataConnectors.json:301]: they are sync.
why would you think they would be async?

**Pull request merged**